SparkPost/2web2ui
# FE-1327 - Upgrades react-query to version 3.5+

[FE-1327](https://sparkpost.atlassian.net/browse/FE-1327)

### What Changed

- Upgrades `react-query` to version 3.5+
- Upgrades `enzyme`-related dependencies that were causing test failures
- Installs `nock` (https://github.com/nock/nock) for stubbing API requests in a Jest DOM environment
- Adds new `stubRequest` test helper that wraps `nock.intercept()`. Very similar API to `cy.stubRequest()`
- Refactors `useSparkPostQuery` by breaking up logic in to re-usable helpers
- Adds Jest unit testing for said helper functions as well as `useSparkPostQuery` and `useSparkPostQueries`
- Refactors `useSparkPostQueries` to more closely align with the newly released [useQueries](https://react-query.tanstack.com/reference/useQueries#_top)

### How To Test

- Verify all features that leverage `useSparkPostQuery` and `useSparkPostQueries` continue to function as expected
- Verify relevant steps from the [React Query v3 migration guide](https://react-query.tanstack.com/guides/migrating-to-react-query-3) were taken
- Does changing the configuration value for `apiBase` in `test-config.js` introduce any breaking changes? Is this just used for our Jest environment or is this used elsewhere?

### To Do

- [x] Cut a follow-up ticket to add `useSparkPostMutation` hook wrapping [useMutation](https://react-query.tanstack.com/reference/useMutation#_top)
